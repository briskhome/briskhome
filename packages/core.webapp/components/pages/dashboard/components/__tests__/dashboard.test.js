import React from 'react';
import toJson from 'enzyme-to-json';
import { mount, render } from 'enzyme';

import { Dashboard } from '../../';

jest.mock('../header', () => () => <div data-mock="DashboardHeader" />);
jest.mock('react-grid-layout', () => ({
  Responsive: jest.fn(props => <div {...props} />),
  WidthProvider: jest.fn(props => props),
}));

describe('core.webapp -> components -> pages -> dashboard -> card', () => {
  it('renders', () => {
    const component = mount(<Dashboard />);
    expect(toJson(render(component))).toMatchSnapshot();
  });

  it('onToggleEditing', () => {
    const component = mount(<Dashboard />);
    component.instance().onToggleEditing();
    expect(toJson(render(component))).toMatchSnapshot();
  });

  it('onLayoutChange', () => {
    const component = mount(<Dashboard />);
    component.instance().onLayoutChange('md', {});
    expect(toJson(render(component))).toMatchSnapshot();
  });
});
