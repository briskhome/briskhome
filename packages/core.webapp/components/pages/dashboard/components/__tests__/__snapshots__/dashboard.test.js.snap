// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core.webapp -> components -> pages -> dashboard -> card onLayoutChange 1`] = `
<div
  class="dashboard"
>
  <div
    data-mock="DashboardHeader"
  />
  <div
    class="dashboard-content"
  >
    <a
      class="dashboard-content__edit-button"
    >
      Edit
    </a>
    <div
      class="layout"
      containerpadding="40,20"
      layouts="[object Object]"
      margin="20,20"
      rowheight="30"
    >
      <div
        class="dashboard-widget__wrapper"
        data-grid="[object Object]"
      />
    </div>
  </div>
</div>
`;

exports[`core.webapp -> components -> pages -> dashboard -> card onToggleEditing 1`] = `
<div
  class="dashboard"
>
  <div
    data-mock="DashboardHeader"
  />
  <div
    class="dashboard-content"
  >
    <a
      class="dashboard-content__edit-button"
    >
      Edit
    </a>
    <div
      class="layout"
      containerpadding="40,20"
      layouts="[object Object]"
      margin="20,20"
      rowheight="30"
    >
      <div
        class="dashboard-widget__wrapper"
        data-grid="[object Object]"
      />
    </div>
  </div>
</div>
`;

exports[`core.webapp -> components -> pages -> dashboard -> card renders 1`] = `
<div
  class="dashboard"
>
  <div
    data-mock="DashboardHeader"
  />
  <div
    class="dashboard-content"
  >
    <a
      class="dashboard-content__edit-button"
    >
      Edit
    </a>
    <div
      class="layout"
      containerpadding="40,20"
      layouts="[object Object]"
      margin="20,20"
      rowheight="30"
    >
      <div
        class="dashboard-widget__wrapper"
        data-grid="[object Object]"
      />
    </div>
  </div>
</div>
`;
